modules/social_features/social_node/src/QueryAccess/SocialNodeEnsureTablesTrait.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\social_node\QueryAccess;
+
+use Drupal\Core\Database\Query\SelectInterface;
+
+/**
+ * Contains helper functions for node entity query alters.
+ */
+trait SocialNodeEnsureTablesTrait {
+
+  /**
+   * Attach a query as object property.
+   *
+   * @param \Drupal\Core\Database\Query\SelectInterface $query
+   *   The query object.
+   */
+  protected function attachQuery(SelectInterface $query): void {
+    $this->query = $query;
+  }
+
+  /**
+   * Ensures the query is joined with a given node field table.
+   *
+   * @return string
+   *   The node field join alias.
+   */
+  public function ensureNodeFieldTableJoin(string $field_name): string {
```

Comment:
I don't like the overall usage of the term `Ensure` here, I don't think it make it very clear/usable but I can't come with another naming so I guess it will be fine :)